### PR TITLE
fix: workaround ansible inconsistent var handling

### PR DESCRIPTION
**Pull request by mhitza**
Context: if a variable is defined within a set_fact module call, and the variable value is used within the new varibles defined via set_fact. If the variable defined already has a value, within the set_fact variables the existing value will be used instead of the newly assigned value. Encounted with Ansible version 2.16.3

Reproduction example:

```yaml
- ansible.builtin.set_fact:
    status: "no a json"

- ansible.builtin.shell: >
     echo "{ \"Self\": { \"Online\": true }, \"Version\": 1.0 }"
  changed_when: false
  register: tailscale_status

- vars:
    status: "{{ tailscale_status.stdout | from_json }}" # <- does not take place
  ansible.builtin.set_fact:
    tailscale_is_online: "{{ status.Self.Online }}"
    tailscale_version: "{{ status.Version }}"
```

---

How I encountered the issue. In a project where the `devsec.hardening` collection -> `os_hardeding` role is used alongside this role, and that roles is included before this role. It defines it's own status variable which triggers a failure case in this role on the changed lines.

In some sense it looks like an unintentional behaviour that it works to use `vars` alongside `set_fact` as it's not listed as part of the attributes it supports https://docs.ansible.com/ansible/latest/collections/ansible/builtin/set_fact_module.html

The same behaviour is relied upon within the handlers, which I've also changed as it was triggering the error.

Upstream issue reported in Ansible https://github.com/ansible/ansible/issues/82698

### CODE REVIEW

**Comment by artis3n** [comment]
This is probably what is causing #421 

I would be in favor of renaming this variable: https://github.com/artis3n/ansible-role-tailscale/issues/421#issuecomment-1910209063

File: handlers/main.yml [hunk 1]
```diff
@@ -7,8 +7,11 @@
...
 - name: Parse status JSON
   listen: Confirm Tailscale is Connected
-  vars:
+  ansible.builtin.set_fact:
     status: "{{ tailscale_status.stdout | from_json }}"
+
+- name: Set facts based on status
+  listen: Confirm Tailscale is Connected
   ansible.builtin.set_fact:
     tailscale_is_online: "{{ status.Self.Online }}"
```
Comment (flowerysong) [inline handlers/main.yml:16]:
```suggestion
  listen: Confirm Tailscale is Connected
  ansible.builtin.set_fact:
    tailscale_is_online: "{{ (tailscale_status.stdout | from_json).Self.Online }}"
```

File: tasks/install.yml [hunk 2]
```diff
@@ -65,8 +65,10 @@
...
 
 - name: Install | Parse status JSON
-  vars:
+  ansible.builtin.set_fact:
     status: "{{ tailscale_status.stdout | from_json }}"
+
+- name: Install | Set facts based on status
   ansible.builtin.set_fact:
     tailscale_is_online: "{{ status.Self.Online }}"
     tailscale_version: "{{ status.Version }}"
```
Comment (flowerysong) [inline tasks/install.yml:74]:
```suggestion
  vars:
    tailscale_status_parsed: "{{ tailscale_status.stdout | from_json }}"
  ansible.builtin.set_fact:
    tailscale_is_online: "{{  tailscale_status_parsed.Self.Online }}"
    tailscale_version: "{{  tailscale_status_parsed.Version }}"
```

**Comment by McSim85** [comment]
> This is probably what is causing #421
> 
> I would be in favor of renaming this variable alongside removing the use of `vars`: [#421 (comment)](https://github.com/artis3n/ansible-role-tailscale/issues/421#issuecomment-1910209063)
> 
> Maybe `tailscale_status` or `artis3n_tailscale_status`?

I would extend this discussion and suggest these changes in the role.
https://github.com/artis3n/ansible-role-tailscale/blob/main/defaults/main.yml  contains variables, that users might want to change. So, they can be used during include_role or import_role.
The `state` and `verbose` variables are pretty common ones and can potentially cause the same issues. 
So, If we rename any variables that do not start with `tailscale_` in https://github.com/artis3n/ansible-role-tailscale/blob/main/defaults/main.yml to the names with `tailscale_`  prefix, this will follow the best practice with [Ansible linter recommendations](https://ansible.readthedocs.io/projects/lint/rules/var-naming/):
- var-naming[no-role-prefix]: Variables names from within roles should use role_name_ as a prefix. Underlines are accepted before the prefix.
- When using include_role or import_role with vars, vars should start with included role name prefix. 

This is breaking changes, though. It might cause backward incompatibility. But in the long term, this will prevent unexpected behavior.

**Comment by mhitza** [comment]
Thanks for the suggestions @flowerysong .

---

Let me know if you'd like to have the `verbose` variable changed in this PR as well or handle that separately.

**Comment by artis3n** [comment]
I'll handle verbose separately, for focused PRs

File: handlers/main.yml [hunk 3]
```diff
@@ -7,10 +7,9 @@
 
 - name: Parse status JSON
   listen: Confirm Tailscale is Connected
-  vars:
-    status: "{{ tailscale_status.stdout | from_json }}"
+  listen: Confirm Tailscale is Connected
```
Comment (artis3n) [inline handlers/main.yml:10]: This line is a duplicate, see the Ansible lint status check failure

**Comment by artis3n** [comment]
Looks like I need to resolve #432 , which just started occurring, I will take a look this weekend.

**Comment by artis3n** [comment]
Please merge from `main` to pick up the CI fix

**Comment by mhitza** [comment]
@artis3n :+1:

**Comment by artis3n** [comment]
Whoops I missed one file (https://github.com/artis3n/ansible-role-tailscale/pull/436), one more merge with main please :)

**Review by artis3n — APPROVED** [review]
Thank you for this PR!

**Comment by mhitza** [comment]
@artis3n any planned new release for the role that includes this fix?

**Comment by artis3n** [comment]
Yes, sorry I am meaning to release it under other structural changes to this role and I've needed to set aside time to get those other things done.

**Comment by artis3n** [comment]
Published a patch release tonight to get this and some other changes out while the larger changes are still in progress.

**Pull request merged**